ndd_slurm.py

```diff
@@ -126,9 +149,15 @@ def get_srun_cmd(args):
     return cmd
 
 
-def init_process(cmd):
-    process = subprocess.Popen(cmd, stdin=subprocess.PIPE)
-    process.stdin.close()
+def init_process(cmd, read_fd=None, write_fd=None):
+    assert read_fd is None or write_fd is None
```

Comment:
That's perfectly correct, but maybe `assert not (read_fd and write_fd)` will be a little more straghtforward to read, so it's up to you.